libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/KaptWithoutKotlincTask.kt

```diff
@@ -225,6 +239,11 @@ private class KaptExecution @Inject constructor(
         val kaptClassLoader = cachedKaptClassLoader ?: URLClassLoader(kaptClasspathUrls, classLoaderWithToolsJar)
         cachedKaptClassLoader = kaptClassLoader
 
+        if (classLoadersProvider == null && classloadersCacheSize > 0) {
+            logger.info("Initializing KAPT classloaders cache with size = $classloadersCacheSize")
+            classLoadersProvider = CachingClassLoadersProvider(classloadersCacheSize, cachedClassLoaderWithToolsJar!!)
```

Comment:
I refactored this part, compiler failed to prove that it can't be null by this point
findRootClassLoader is here https://github.com/JetBrains/kotlin/pull/4176/files#diff-358e307c9ea5ec880f024401af1b0f1f81e24ca92484a78a3c7e53c62b882d6cR313